src/lib/FlightTasks/tasks/Utility/ObstacleAvoidance.hpp

```diff
@@ -116,6 +116,8 @@ class ObstacleAvoidance : public ModuleParams
 	orb_advert_t _pub_vehicle_command{nullptr}; /**< vehicle command do publication */
 
 	matrix::Vector3f _curr_wp = {}; /**< current position triplet */
+	matrix::Vector3f _position = {}; /**< current position triplet */
```

Comment:
```suggestion
	matrix::Vector3f _position = {}; /**< current vehicle position */
```

Reply:
You might want to have the `avoidance_data_timeout` and `avoidance_point_valid` checks done first on any loop iteration for avoidance and then have a member `_avoidance_setpoint_available` which you can also check in other methods such that you don't have to duplicate the position state but that's just architectural preference.

Reply:
@MaEtUgR  What do you mean by duplicate position state?

Reply:
Was changed
@mrivi You don't necessarily need to fix it now but I saw it as not so nice that you have to cache the current vehicle position just because it's available in a different method than the avoidance available flag(s).